src/coreclr/jit/gentree.cpp

```diff
@@ -12463,7 +12584,7 @@ void Compiler::gtDispLIRNode(GenTree* node, const char* prefixMsg /* = nullptr *
     IndentInfo operandArc = IIArcTop;
     for (GenTree* operand : node->Operands())
     {
-        if (!operand->IsValue())
+        if (!operand->IsValue() && (operand->gtFlags & GTF_SET_FLAGS) == 0)
```

Comment:
Why this change? It does not seem right to skip an operand just because it has this flag set.

Reply:
The code is already being skipped if the node does not generate a value. I'm adding an exception to that case so that it doesn't skip if GTF_SET_FLAGS is set.

Why?

In this code we don't want the EQ to generate a value. So it's set to void and GTF_SET_FLAGS is set.
```
               [000592] -----------                         \--*  SELECT    ref   
               [000589] N------N---                            +--*  EQ        int   
               [000590] -----------                            |  +--*  LCL_VAR   ref    V10 tmp7         
               [000591] -----------                            |  \--*  CNS_INT   ref    null
               [000412] -----+-----                            +--*  CNS_INT   ref    null
               [000588] -----------                            \--*  LCL_VAR   ref    V34 tmp31        

```

So, when lower IR is generated the EQ without my change, the EQ 589 node wouldn't have been dumped:
```
                                                            /--*  t589   void   
                                                            +--*  t412   ref    
                                                            +--*  t588   ref    
N537 ( 10,  8) [000592] -----------                  t592 = *  SELECT    ref    REG x0 $94
```

Reply:
Ah, I misread the code, this is just displaying the edges when the operand does set the flag. Seems fine to me (though having the edges in the first place is unfortunate, see my comment below).

Reply:
This change has been removed.